litex/soc/software/bios/cmds/cmd_bios.c

```diff
@@ -8,10 +8,61 @@
 #include <sim_debug.h>
 
 #include <generated/csr.h>
-
+#include <generated/mem.h>
 #include "../command.h"
 #include "../helpers.h"
 
+
+#define DOT_VAL	0x80
+
+static unsigned char hex_pos[8] = {
+	0x01,
+	0x02,
+	0x04,
+	0x08,
+	0x10,
+	0x20,
+	0x40,
+	0x80
+};
+
+static unsigned char hex_val[16]  = {
+	(0x01 | 0x02 | 0x04 | 0x08 | 0x10 | 0x20),        // 0
+	(0x02 | 0x04),                                    // 1
+	(0x01 | 0x02 | 0x40 | 0x10 | 0x08),               // 2
+	(0x01 | 0x02 | 0x04 | 0x08 | 0x40),               // 3
+	(0x02 | 0x04 | 0x20 | 0x40),                      // 4
+	(0x01 | 0x04 | 0x20 | 0x40 | 0x08),               // 5
+	(0x01 | 0x04 | 0x08 | 0x10 | 0x20 | 0x40),        // 6
+	(0x20 | 0x01 | 0x02 | 0x04),                      // 7
+	(0x01 | 0x02 | 0x04 | 0x08 | 0x10 | 0x20 | 0x40), // 8
+	(0x01 | 0x02 | 0x04 | 0x08 | 0x20 | 0x40),        // 9
+	(0x01 | 0x02 | 0x04 | 0x10 | 0x20 | 0x40),        // A
+	(0x04 | 0x08 | 0x10 | 0x20 | 0x40),               // B
+	(0x01 | 0x08 | 0x10 | 0x20),                      // C
+	(0x02 | 0x04 | 0x08 | 0x10 | 0x40),               // D
+	(0x01 | 0x08 | 0x10 | 0x20 | 0x40),               // E
+	(0x01 | 0x10 | 0x20 | 0x40),                      // F
+};
+
```

Comment:
This code seems related to the seven segment, so should probably not be in cmd_bios.

Reply:
Dear Sir:

I think what you talk about is right. could I withdraw this PR ? 
after put the code into where you suggest and send PR again?

BR, Sanada